src/FSharp.Markdown/MarkdownParser.fs

```diff
@@ -145,107 +146,128 @@ let (|Emphasised|_|) = function
     | _ -> None
   | _ -> None
 
+/// Defines a context for the main `parseParagraphs` function
+type ParsingContext =
+  { Links : Dictionary<string, string * option<string>>
+    Newline : string
+    CurrentRange : MarkdownRange option }
+
 /// Parses a body of a paragraph and recognizes all inline tags.
-let rec parseChars acc input = seq {
+let rec parseChars acc input (ctx:ParsingContext) = seq {
 
   // Zero or one literals, depending whether there is some accumulated input
   let accLiterals = Lazy.Create(fun () ->
-    if List.isEmpty acc then []
-    else [Literal(String(List.rev acc |> Array.ofList))] )
+    if List.isEmpty acc then ([], ctx)
+    else ([Literal(String(List.rev acc |> Array.ofList), match ctx.CurrentRange with | Some(n) -> Some({ n with EndColumn = n.StartColumn + acc.Length }) | None -> None)], { ctx with CurrentRange = match ctx.CurrentRange with | Some(n) -> Some({ n with StartColumn = n.StartColumn + acc.Length }) | None -> None }) )
```

Comment:
This looks like a super-long and hard to parse line - could you refactor it into something a bit more verbose, but a bit more readable?